nxdrive/gui/folders_dialog.py

```diff
@@ -402,10 +405,20 @@ def _process_additionnal_local_paths(self, paths: List[str]) -> None:
 
     def _select_more_files(self) -> None:
         """Choose additional local files to upload."""
-        paths, _ = QFileDialog.getOpenFileNames(self, Translator.get("ADD_FILES"))
+        paths, _ = QFileDialog.getOpenFileNames(
+            self, Translator.get("ADD_FILES"), self.local_folder_path
+        )
+        if paths:
+            self.engine.dao.update_config(
+                "dt_last_local_selected_location", Path(paths[0]).parent
+            )
```

Comment:
Please move that logic on the `Engine` side, as it is done for other database fields.